### PR TITLE
Support Comments in SQL

### PR DESCRIPTION
**Pull request by andytill**
Comments come in two varieties in SQL, the multiline kind:

```sql
/* a multiline comment
   same as in c++ or java
*/
```
Or a single line comment, like an `%` in erlang:
```sql
SELECT * FROM mytable; -- this is my comment
```
Implementing this now because some of our example SQL snippets in the docs do not parse because of comment syntax we do not support.

### CODE REVIEW

**Comment by paegun** [comment]
+1

File: src/riak_ql_lexer.xrl [hunk 1]
```diff
@@ -46,6 +46,7 @@ REGEX = (/[^/][a-zA-Z0-9\*\.]+/i?)
 
 IDENTIFIER = ([a-zA-Z][a-zA-Z0-9_\-]*)
 QUOTED_IDENTIFIER = \"(\"\"|[^\"\n])*\"
+COMMENT_MULTILINE = (/\*([^*]|[\r\n]|(\*+([^*/]|[\r\n])))*\*+/)|(--.*)
```
Comment (hmmr) [inline src/riak_ql_lexer.xrl:49]: I could have it capture the comments with a slightly simplified `(/\*([^*]|[\r\n]|(\*([^/]|[\r\n])))*\*/)|(--.*)`.

Comment (andytill) [reply src/riak_ql_lexer.xrl:49]: Can you explain the change?

Comment (paegun) [reply src/riak_ql_lexer.xrl:49]: I similarly evaluated the C-style block comment regex and found it not optimal, but accurate according to the tests included and a few additional test cases that stress the interweaving of C-style block comment and SQL-style end-of-line comment, which are apparently covered by the regex and proved out to be. With that in mind, I did not feel it necessary to suggest a change to the regex.

The difference between the regexex ^^,
1.

``` javascript
submitted: (/\*([^*]|[\r\n]|(\*+([^*/]|[\r\n])))*\*+/)|(--.*)
suggested: (/\*([^*]|[\r\n]|(\*([^/]|[\r\n])))*\*/)|(--.*)
______________________________^__ multiple asterisks don't matter, they should be gobbled up by being w/i the C-style block comment
```

2.

``` javascript
submitted:  ...([^*/]|[\r\n])))*\*+/)|(--.*)
suggested: ...([^/]|[\r\n])))*\*/)|(--.*)
_________________^__ the end-of-comment-block asterisk slash is specific enough to allow for not needing to specify not asterisk in the body of the comment block
```

3.

``` javascript
submitted:  ...\*+/)|(--.*)
suggested: ...\*/)|(--.*)
________________^__ multiple asterisks don't matter, they should be gobbled up by being w/i the C-style block comment
```

Comment (hmmr) [reply src/riak_ql_lexer.xrl:49]: Thanks @paegun for picking it apart and explaining!

**Comment by hmmr** [comment]
+1

**Pull request merged**